plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java

```diff
@@ -910,6 +912,255 @@ public void testSettingChangeDataFeedEnabledProperty()
+                "THEN DELETE " +
+                "WHEN NOT MATCHED " +
+                "THEN INSERT (page_url, domain, views) VALUES (source.page_url, source.domain, source.views)", 4);
+
+        assertQuery("SELECT * FROM " + targetTable,
+                "VALUES('url3', 'domain3', 3), ('url4', 'domain1', 44), ('url5', 'domain3', 5)");
+
+        MaterializedResult result = computeActual("SELECT * FROM TABLE(system.table_changes('test_schema', '" + targetTable + "'))");
+        List<MaterializedRow> resultWithoutTimestamp = getCdfColumnsWithoutTimestamp(result);
+        assertThat(resultWithoutTimestamp).containsOnly(
+                new MaterializedRow(ImmutableList.of("url1", "domain1", 1, "insert", 1)),
+                new MaterializedRow(ImmutableList.of("url2", "domain2", 2, "insert", 1)),
+                new MaterializedRow(ImmutableList.of("url3", "domain3", 3, "insert", 1)),
+                new MaterializedRow(ImmutableList.of("url4", "domain1", 4, "insert", 1)),
+                new MaterializedRow(ImmutableList.of("url1", "domain1", 1, "delete", 2)),
+                new MaterializedRow(ImmutableList.of("url2", "domain2", 2, "delete", 2)),
+                new MaterializedRow(ImmutableList.of("url4", "domain1", 4, "update_preimage", 2)),
+                new MaterializedRow(ImmutableList.of("url4", "domain1", 44, "update_postimage", 2)),
+                new MaterializedRow(ImmutableList.of("url5", "domain3", 5, "insert", 2)));
+
+        String sourceTable2 = "test_basic_operations_on_partitioned_table_with_cdf_enabled_source_1_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + sourceTable2 + " (page_url VARCHAR, domain VARCHAR, views INTEGER)");
+        assertUpdate("INSERT INTO " + sourceTable2 +
+                " VALUES('url3', 'domain1', 300), ('url4', 'domain2', 400), ('url5', 'domain3', 500), ('url6', 'domain1', 600)", 4);
+
+        assertUpdate("MERGE INTO " + targetTable + " target USING " + sourceTable2 + " source " +
+                "ON (target.page_url = source.page_url) " +
+                "WHEN MATCHED AND target.views > 3 " +
+                "THEN UPDATE SET domain = source.domain, views = (source.views + target.views) " +
+                "WHEN MATCHED AND target.views <= 3 " +
+                "THEN DELETE " +
+                "WHEN NOT MATCHED " +
+                "THEN INSERT (page_url, domain, views) VALUES (source.page_url, source.domain, source.views)", 4);
+
+        assertQuery("SELECT * FROM " + targetTable,
+                "VALUES('url4', 'domain2', 444), ('url5', 'domain3', 505), ('url6', 'domain1', 600)");
+
+        result = computeActual("SELECT * FROM TABLE(system.table_changes('test_schema', '" + targetTable + "', 2))");
+        resultWithoutTimestamp = getCdfColumnsWithoutTimestamp(result);
+        assertThat(resultWithoutTimestamp).containsOnly(
+                new MaterializedRow(ImmutableList.of("url3", "domain3", 3, "delete", 3)),
```

Comment:
Use a helper method `MaterializedRow row(Object... values)` to make the assertions easier to read.

Reply:
Where is that method? I can't fint it

Reply:
Sorry for the confusion. I was hinting about introducing such a method :) to be more concise in the assertions.